### PR TITLE
Add Deposit resource and Operations

### LINKED ISSUE #22: Add resource support for Deposit (3 hrs)
https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/deposit

We need to be able to create, update, read, and delete deposits, including using Searchers for this

Attribute-wise, we need the following
- ID 
- DepositToAccountRef
- Line
- CurrencyRef
- PrivateNote
- ExchangeRate
- TxnDate

### PR DESCRIPTION
**Pull request by jozefvaclavik**
Fix #22

### CODE REVIEW

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/create.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                'Line' => resource.line_items.map do |line_item|
+                  {
+                    'DetailType' => 'DepositLineDetail',
+                    'DepositLineDetail' => {
+                      'AccountRef' => {
+                        'value' => line_item.account&.ledger_id || resource.to_account.ledger_id
+                      }
+                    },
+                    'Amount' => line_item.amount / 100.0,
+                    # 'Description' => line_item.description
```
Comment (mattgmarcus) [inline lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/create.rb]: @jozefvaclavik why is this commented out

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/searcher.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+              adaptor
+                .query(
+                  resource: 'deposit',
+                  limit: limit,
+                  offset: offset
+                )
+                .map do |c|
+                  LedgerSync::Deposit.new(
+                    ledger_id: c.fetch('Id'),
+                    name: c.dig('Id')
```
Comment (mattgmarcus) [inline lib/ledger_sync/adaptors/quickbooks_online/deposit/searcher.rb]: Is this a typo? Or are you just saying the ID is the name?

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          class Find < Operation::Find
+            class Contract < LedgerSync::Adaptors::Contract
+              schema do
+                required(:ledger_id).filled(:string)
+                required(:to_account).hash(Types::Reference)
+                required(:currency).maybe(:string)
+                required(:memo).maybe(:string)
+                required(:transaction_date).filled(:date?)
+                required(:exchange_rate).maybe(:float)
+                required(:line_items).array(Types::Reference)
```
Comment (mattgmarcus) [inline lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb]: @jozefvaclavik - it looks like there's a `TotalAmt` field in the API that is read only. Can we maybe fetch that and expose it as an attribute you can read? There's no way to write to it, so we wouldn't let you set it in the create operation

Comment (ryanwjackson) [reply lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb]: @mattgmarcus I'm not sure it's that easy.  The best way to get that info is  `result.operation.response` if it is not an attribute of the resource.  In my serializer PR, I am making it so it merges in any other values on create/update/find as opposed to just setting the ID or even leaving it as what we passed in.

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/searcher.rb (hunk 2 above)
Comment (jozefvaclavik) [reply lib/ledger_sync/adaptors/quickbooks_online/deposit/searcher.rb]: Actually in this case it doesnt make sense to set name. As thats not an attribute. Name is computed property.

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb (hunk 3 above)
Comment (jozefvaclavik) [reply lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb]: Yea, once we have serialization, we can do that.

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/create.rb (hunk 1 above)
Comment (jozefvaclavik) [reply lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/create.rb]: Ups, sigh.

File: lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb (hunk 3 above)
Comment (mattgmarcus) [reply lib/ledger_sync/adaptors/quickbooks_online/deposit/operations/find.rb]: Kk sounds good

**Pull request merged**